[docs][data grid] Revise the Row Grouping doc (#16217)
mui/mui-x: docs/data/data-grid/server-side-data/tree-data.md

## Group expansion

> Group expansion of server-side tree data works the same as explained in [Row grouping—Group expansion](/x/react-data-grid/row-grouping/#group-expansion).

Reviewer: Change the wording a bit since it's not "exactly" same as mentioned in the next line?
```suggestion
Group expansion of server-side tree data works similar to the [Row grouping—Group expansion](/x/react-data-grid/row-grouping/#group-expansion).
```

Author: ```suggestion
Group expansion of server-side tree data works similarly to how it's described in [Row grouping—Group expansion](/x/react-data-grid/row-grouping/#group-expansion).
```
